### PR TITLE
don’t rely on file listing to check file exists

### LINKED ISSUE #250: Importer won't find data.yaml if there are too many files in the S3 bucket
When `DataMagic::Config` looks for `data.yaml` it grabs the list of files in the S3 bucket and searches it for the filename. However, `@s3.list_objects()` only returns 1000 objects at a time, and if there are over 1000 objects in the bucket (because of backups/snapshots) then `data.yaml` won't be in that list and won't be loaded.

I suspect the best solution here is to not rely on file listing, and instead just assume that `data.yaml` is there and try to read it, taking an alternate path if that fails.

### PR DESCRIPTION
**Pull request by ultrasaurus**
S3 library only fetches 1000 keys, 
so file_list method would fail when directory had lots of files
easy fix removes file_list method entirely, just attempts to read
file and deals with failure

resolves #250

### CODE REVIEW

File: lib/data_magic/config.rb [hunk 1]
```diff
@@ -317,32 +321,19 @@ def read_path(path)
...
-    end
-
     def load_yaml(path = nil)
       logger.info "load_yaml: #{path}"
-      file = data_file_name(path)
-      if file.nil? and not ENV['ALLOW_MISSING_YML']
-        logger.warn "No data.y?ml found; using default options"
+      raw = read_path(File.join(path, "data.yaml"))
+      if raw.empty?
+        raw = read_path(File.join(path, "data.yml"))
```
Comment (yozlet) [inline lib/data_magic/config.rb]: would 
```ruby
raw = read_path(File.join(path, "data.yaml")) || read_path(File.join(path, "data.yml"))
```
... work here?

**Comment by yozlet** [comment]
Looks good, though there are two related issues:

1. Failure to load config should cause the app to loudly exit
2. If `read_path()` fails, it should find & log the cause of failure

However, both of these can be done elsewhere.

**Comment by ultrasaurus** [comment]
#1 https://github.com/18F/open-data-maker/issues/254
#2 https://github.com/18F/open-data-maker/issues/255

File: lib/data_magic/config.rb (hunk 1 above)
Comment (ultrasaurus) [inline lib/data_magic/config.rb]: the OR won't work because it is empty not nil

**Comment by yozlet** [comment]
:+1:

**Pull request merged**